### PR TITLE
Add details of ‘confirm email address’ and ‘custom retention period’

### PR DESCRIPTION
**Pull request by karlchillmaid**
# Do not merge until the new features are live

This PR updates the ‘Send a file by email’ section of the REST-API documentation.

There are 2 stories in the backlog which describe this work:

* [Update 'send a file by email' API client docs for 'email verification' flow](https://www.pivotaltracker.com/story/show/182942256)
* [Update 'send a file by email' API client docs for custom retention periods](https://www.pivotaltracker.com/story/show/182999152)

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -385,6 +389,63 @@ Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
...
+
+byte[] documentContents = File.ReadAllBytes("<file path>");
+
+Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
+{
+    { "name", "Foo" },
+    { "link_to_file", NotificationClient.PrepareUpload(documentContents)}
+};
+```
+
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:448]: Not sure why the .NET client is missing `Response` section for ‘Send a file by email’.

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -385,6 +389,63 @@ Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
...
+```csharp
+
+byte[] documentContents = File.ReadAllBytes("<file path>");
+
+Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
+{
+    { "name", "Foo" },
+    { "link_to_file", NotificationClient.PrepareUpload(documentContents)}
+};
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:410]:
```suggestion
    { "name", "Foo" },
    { "link_to_file", NotificationClient.PrepareUpload(documentContents)},
    { "verify_email_before_download", true}
};
```

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -385,6 +389,63 @@ Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
...
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:427]:
```suggestion
{
    { "name", "Foo" },
    { "link_to_file", NotificationClient.PrepareUpload(documentContents)},
    { "verify_email_before_download", false}
};
```

File: DOCUMENTATION.md [hunk 4]
```diff
@@ -385,6 +389,63 @@ Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
...
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:446]:
```suggestion
    { "name", "Foo" },
    { "link_to_file", NotificationClient.PrepareUpload(documentContents)},
    { "retention_period", 4 weeks}
};
```

File: DOCUMENTATION.md [hunk 5]
```diff
@@ -385,6 +392,78 @@ Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
...
+
+This new security feature is optional. You should use it if you send files that are sensitive - for example, because they contain personal information about your users.
+
+When a recipient clicks the link in the email you’ve sent them, they have to enter their email address. Only someone who knows the recipient’s email address can download the file.
+
+From 29 March 2023, we will turn this feature on by default for every file you send. Files sent before 29 March will not be affected.
+
+##### Turn on email address check
+
+To use this feature before 29 March 2023 you will need version X.X.X of the Python client library, or a more recent version.
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:405]:
```suggestion
To use this feature before 29 March 2023 you will need version X.X.X of the .NET client library, or a more recent version.
```

File: DOCUMENTATION.md [hunk 6]
```diff
@@ -385,6 +392,80 @@ Dictionary<String, dynamic> personalisation = new Dictionary<String, dynamic>
...
+};
+```
+
+#### Choose the length of time that a file is available to download
+
+Set the number of weeks you want the file to be available using the `retention_period` key.
+
+You can choose any value between 1 week and 78 weeks.
+
+If you do not do this, the file will be available for the default period of 78 weeks (18 months).
```
Comment (karlchillmaid) [inline DOCUMENTATION.md:456]:
```suggestion
To use this feature will need version X.X.X of the .NET client library, or a more recent version.

If you do not choose a value, the file will be available for the default period of 78 weeks (18 months).
```

**Comment by samuelhwilliams** [comment]
Merged content into https://github.com/alphagov/notifications-net-client/pull/164

**Pull request closed without merging**